core/libraries/batch/JobHandlers/RegistrationsReport.php

```diff
@@ -506,12 +498,74 @@ public function get_csv_data_for($event_id, $offset, $limit, $question_labels, $
                     );
                 }
             }
-            // Include check-in data
-            if ($incude_checkins) {
+            // Include check-in data for datetime
+            if ($event_id && $DTT_id) {
+                // get whether or not the user has checked in
+                $reg_csv_array[ (string) esc_html__('Datetime Check-ins #', 'event_espresso') ] = $reg_model->count_related(
+                    $reg_row['Registration.REG_ID'],
+                    'Checkin',
+                    [
+                        [
+                            'DTT_ID' => $DTT_id
+                        ]
+                    ]
+                );
+                $datetime = EEM_Datetime::instance()->get_one_by_ID($DTT_id);
+                $checkin_rows = EEM_Checkin::instance()->get_all_wpdb_results(
+                    [
+                        [
+                            'REG_ID' => $reg_row['Registration.REG_ID'],
+                            'DTT_ID' => $datetime->get('DTT_ID'),
+                        ],
+                    ]
+                );
+                if (trim($datetime->get('DTT_name'))) {
+                    /* translators: 1: datetime name, 2: datetime ID */
+                    $datetime_name = sprintf(
+                        esc_html__('Check-ins for %1$s - ID: %2$s', 'event_espresso'),
+                        esc_html($datetime->get('DTT_name')),
+                        esc_html($datetime->get('DTT_ID'))
+                    );
+                } else {
+                    /* translators: %s: datetime ID */
+                    $datetime_name = sprintf(
+                        esc_html__('ID: %1$s', 'event_espresso'),
+                        esc_html($datetime->get('DTT_ID'))
+                    );
+                }
+                $checkins = [];
+                foreach ($checkin_rows as $checkin_row) {
+                    if ((int) $checkin_row['Checkin.CHK_in'] === 1) {
+                        /* translators: 1: check-in status, 2: check-in timestamp */
+                        $checkin_value = sprintf(
+                            esc_html__('%1$s: %2$s', 'event_espresso'),
+                            esc_html__('IN', 'event_espresso'),
```

Comment:
don't need/want nested escaping, so this would be better as:

```php
                            esc_html__('%1$s: %2$s', 'event_espresso'),
                            __('IN', 'event_espresso'),
```

I'm also on the fence as to whether the following would be better:

```php
                        $checkin_value = sprintf(
                        	/* translators: 1: check-in timestamp */
                            esc_html__('IN: %1$s', 'event_espresso'),
                            date('Y-m-d g:i a', strtotime($checkin_row['Checkin.CHK_timestamp']))
                        );
```

Reply:
I did the first approach since the translated IN and OR texts were totally unnecessary and they could be merged into the first `esc_html__(` function.

    esc_html__('IN: %1$s', 'event_espresso'),

I believe the translators are still able to change `: ` characters to something else.

Reply:
ya as long as that is part of the string passed to one of the i18n functions then they can change it. 
on a tangentially related point, it's also super important to always use numbered placeholders like `%1$s` & `%2$s` as opposed to just `%s`, because it gives translators the freedom to rearrange the order of the placeholders, but you likely already knew that 😉